scipy/special/_extra_special.h

```diff
@@ -0,0 +1,13 @@
+#pragma once
+
+#include <complex>
+#include <numpy/npy_math.h>
+
+#include "_lambertw.h"
+
+
+inline npy_cdouble lambertw_scalar(npy_cdouble zp, long k, double tol) {
```

Comment:
I prefer the reinterpret_cast version, but it's not a big deal and we can also just leave it for later. The reason why I prefer the `reinterpret_cast` version is that it can be implemented in a single line, even if there are multiple arguments that need to be cast. I do think that's the C++ way of doing things.

But also it's not a big deal, and I suspect we'll be modifying these helpers later anyway.

Reply:
Yeah, I like `reinterpret_cast` better too. As you saw below, the issue is that in some tests we build with `-Werror -Wstrictaliasing`, so using `reintrepret_cast` breaks the build.

Reply:
Yep, I saw! What we can think about the best way to do this, is we'll have to do it again and again for all functions we wrap. What I'd suggest is we come up with our own `bit_cast` until C++20 is workable (a long time), but can do that in a later PR.